src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToRunCrossgen.cs

```diff
@@ -0,0 +1,257 @@
+                EnvironmentInfo.GetCompatibleRid(targetFramework));
+
+            var mainProjectDll = Path.Combine(publishDirectory.FullName, $"{projectName}.dll");
+            var classLibDll = Path.Combine(publishDirectory.FullName, $"Classlib.dll");
+
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                publishDirectory.Should().HaveFiles(new[] {
+                    GetPDBFileName(mainProjectDll),
+                    GetPDBFileName(classLibDll),
+                });
+            }
+
+            DoesImageHaveR2RInfo(mainProjectDll).ToString().Should().Be(true.ToString());
+            DoesImageHaveR2RInfo(classLibDll).ToString().Should().Be(true.ToString());
+        }
+
+        [Theory]
+        [InlineData("netcoreapp3.0")]
+        public void It_does_not_support_cross_platform_readytorun_compilation(string targetFramework)
+        {
+            var ridToUse = EnvironmentInfo.GetCompatibleRid(targetFramework);
+            int separator = ridToUse.LastIndexOf('-');
+
+            string platform = ridToUse.Substring(0, separator).ToLowerInvariant();
+            string architectureStr = ridToUse.Substring(separator + 1).ToLowerInvariant();
+
+            var testProject = new TestProject()
+            {
+                Name = "FailingToCrossgen",
+                TargetFrameworks = "netcoreapp3.0",
+                IsExe = true,
+                IsSdkProject = true,
+            };
+
+            if (platform.Contains("win"))
+            {
+                testProject.RuntimeIdentifier = $"linux-{architectureStr}";
+            }
+            else if ((platform.Contains("linux") || platform.Contains("freebsd")) && !platform.Contains("linux-musl"))
+            {
+                testProject.RuntimeIdentifier = $"linux-musl-{architectureStr}";
+            }
+            else if (platform.Contains("linux-musl") || platform.Contains("alpine"))
+            {
+                testProject.RuntimeIdentifier = $"linux-{architectureStr}";
+            }
+            else if (platform.Contains("osx"))
+            {
+                testProject.RuntimeIdentifier = $"win-{architectureStr}";
+            }
+            else if (platform.Contains("rhel.6"))
+            {
+                testProject.RuntimeIdentifier = $"linux-{architectureStr}";
+            }
+            else
+            {
+                return;
+            }
+
+            testProject.AdditionalProperties["ReadyToRun"] = "True";
+
+            var testProjectInstance = _testAssetsManager.CreateTestProject(testProject)
+                .Restore(Log, testProject.Name);
+
+            var publishCommand = new PublishCommand(Log, Path.Combine(testProjectInstance.Path, testProject.Name));
+
+            publishCommand.Execute("/v:n")
+                .Should()
+                .Fail()
+                .And.HaveStdOutContainingIgnoreCase("not supported for the selected target platform or architecture");
```

Comment:
We should not rely on English messages here. Check for the error code.

Reply:
How can I check for the error? Should I check for "NETSDKnnnn" instead?

Reply:
Translated error messages look like they still have the "NETSDKnnnn" text. I think i can check with for that, unless you can think of a reason why this wouldn't necessary work?

Reply:
That's exactly what I meant by check the error code. :)